dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -312,11 +316,9 @@ public void scan() throws CDIException {
 
             //and pull out the corresponding classes
             for (String className : rawBeanclassNames) {
-                Class<?> clazz = this.allClasses.get(className);
-                if (clazz != null) {
-                    if (clazz.getClassLoader() == classLoader || !isAccessibleBean(clazz)) {
-                        this.beanClasses.put(className, clazz);
-                    }
+
+                if (this.allClasses.contains(className)) {
```

Comment:
Isn't `this.allClasses` still empty at this point? Does `this.allClasses` ever get fully populated?
p.s. hard to find things in this view!